packages/job-manager/src/worker/worker.ts

```diff
@@ -51,16 +51,29 @@ export class Worker {
 
             this.poolLoadGenerator.setLastTasksIncrementCount(tasksQueuedCount);
 
+            const runningTasks = poolMetricsInfo.load.runningTasks;
+            const samplingIntervalInSeconds = this.poolLoadGenerator.samplingIntervalInSeconds;
+            const maxParallelTasks = poolMetricsInfo.maxTasksPerPool;
+
             this.logger.logInfo('Pool load statistics', {
                 activeTasks: poolMetricsInfo.load.activeTasks.toString(),
-                runningTasks: poolMetricsInfo.load.runningTasks.toString(),
+                runningTasks: runningTasks.toString(),
                 requestedTasksToAddPerInterval: tasksIncrementCount.toString(),
                 tasksAddedPerInterval: tasksQueuedCount.toString(),
-                samplingIntervalInSeconds: this.poolLoadGenerator.samplingIntervalInSeconds.toString(),
+                samplingIntervalInSeconds: samplingIntervalInSeconds.toString(),
                 processingSpeedTasksPerMinute: this.poolLoadGenerator.processingSpeedPerMinute.toString(),
                 activeToRunningTasksRatio: this.poolLoadGenerator.activeToRunningTasksRatio.toString(),
             });
 
+            const batchPoolMeasurements: BatchPoolMeasurements = {
+                runningTasks,
+                samplingIntervalInSeconds,
+                maxParallelTasks,
+            };
+
+            // tslint:disable-next-line: no-null-keyword
+            this.logger.trackEvent('ScanTasksQueued', null, batchPoolMeasurements);
```

Comment:
you could add strict typings for event content.

something like this:

`
declare type TelemetryEvents = {
    scanQueued: { runningTasks: number };
    scanCompleted: { duration: number };
    eventWithoutData: void;
};

function trackEvent<K extends keyof TelemetryEvents>(eventName: K, eventValues: TelemetryEvents[K]): void {}

trackEvent('scanCompleted', { duration: 10 });
trackEvent('eventWithoutData', undefined);
`

Reply:
Updated. Except that for eventWithoutData the type need to be ``null`` instead of ``void``.